### PR TITLE
Implement return values for enhanced UpdateItem

### PR DESCRIPTION
**Pull request by shetsa-amzn**
Adds support for specifying ReturnValues in UpdateItemEnhancedRequest. Defaults to the current behavior ALL_NEW.

## Motivation and Context
https://github.com/aws/aws-sdk-java-v2/issues/4247

## Testing
Adding Units to Cover all the options of ReturnValue including validating the behavior when its not set.

## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)

## Checklist

- [ x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ x] Local run of `mvn install` succeeds
- [ x] My code follows the code style of this project
- [ x] My change requires a change to the Javadoc documentation
- [ x] I have updated the Javadoc documentation accordingly
- [ x] I have added tests to cover my changes
- [ x] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by davidh44** [comment]
Can we add end-to-end tests as well?
https://github.com/aws/aws-sdk-java-v2/tree/master/services-custom/dynamodb-enhanced/src/it/java/software/ama

**Comment by shetsa-amzn** [comment]
> Can we add end-to-end tests as well? https://github.com/aws/aws-sdk-java-v2/tree/master/services-custom/dynamodb-enhanced/src/it/java/software/ama

Added Integration tests to the second revision.

**Review by davidh44 — APPROVED** [review]

**Comment by davidh44** [comment]
Looks good, approved, can we add a changelog entry?
https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md#pull-request-readiness

**Comment by shetsa-amzn** [comment]
> Looks good, approved, can we add a changelog entry? https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md#pull-request-readiness

Added. Thanks

File: .changes/next-release/feature-DynamoDBEnhancedClient-0c57ace.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "category": "DynamoDB Enhanced Client", 
+    "contributor": "shetsa-amzn", 
+    "type": "feature", 
+    "description": "Adding a member to the UpdateItemEnhancedRequest.Builder named returnItems, that accepts values from the com.amazonaws.services.dynamodbv2.model.ReturnValue enum, and defaults to the current behaviour ALL_NEW, and pass it to the request instead of the constant ALL_NEW."
```
Comment (davidh44) [inline .changes/next-release/feature-DynamoDBEnhancedClient-0c57ace.json:5]: Can we simplify description to something like:
Adds support for specifying ReturnValue in UpdateItemEnhancedRequest

And update the PR description as well?
com.amazonaws.services.dynamodbv2.model is v1 actually

**Pull request merged**